Polish distributed autograd docs.

pritamdamania87 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#29942 Polish distributed autograd docs.**

1) Added links to the design.
2) Fixed function signautres.
3) Expanded examples

Differential Revision: [D18547103](https://our.internmc.facebook.com/intern/diff/D18547103/)

Commit:
Polish distributed autograd docs.

1) Added links to the design.
2) Fixed function signautres.
3) Expanded examples

Differential Revision: [D18547103](https://our.internmc.facebook.com/intern/diff/D18547103/)

[ghstack-poisoned]

rohan-varma commented on docs/source/rpc.rst:
```diff
@@ -58,8 +58,7 @@ used for applications such as model parallel training. In short, applications
 may send and receive gradient recording tensors over RPC. In the forward pass,
 we record when gradient recording tensors are sent over RPC and during the
 backward pass we use this information to perform a distributed backward pass
-using RPC. For more details see the design doc
```
You may want to rebase to avoid merge conflicts as this is different in the latest master (it was updated to refer to the note when that PR got merged)

rohan-varma reviewed (approved):
Thanks for improving our documentation!

Commit:
Update on "Polish distributed autograd docs."


1) Added links to the design.
2) Fixed function signautres.
3) Expanded examples

Differential Revision: [D18547103](https://our.internmc.facebook.com/intern/diff/D18547103/)

[ghstack-poisoned]
